src/common/options/rgw.yaml.in

```diff
@@ -59,6 +59,14 @@ options:
   services:
   - rgw
   with_legacy: true
+- name: rgw_parquet_buffer_size
+  type: size
+  level: advanced
+  desc: the Maximum parquet buffer size, a limit to memory consumption for parquet reading operations.
+  default: 1_G
```

Comment:
this seems way too high for the default. rgw is willing to process rgw_max_concurrent_requests=1024 of these requests at once. normal ops like Put/GetObject only buffer up to ~16MB each

is this much buffering really necessary to get reasonable performance?

Reply:
> this seems way too high for the default. rgw is willing to process rgw_max_concurrent_requests=1024 of these requests at once. normal ops like Put/GetObject only buffer up to ~16MB each
> 
> is this much buffering really necessary to get reasonable performance?


examining this issue.
as mentioned the the tradeoff is the number of calls to s3-storage (its internal calls, s3select-engine -> RGW::range_request)

from a few tests i did there is a difference in the number of calls(obviously), i'm not sure as for the total-latencies

taking into account the parallel calls, maybe it is worth considering a "more dynamic" system 
i.e. as the number of s3select-requests increases the max-buffer-size goes down.

Reply:
> as mentioned the the tradeoff is the number of calls to s3-storage (its internal calls, s3select-engine -> RGW::range_request)

that calls into `RGWGetObj::execute()` which is streaming 4MB rados objects to satisfy the requested range. requesting larger ranges won't reduce the number of osd requests